Wikipedia/UI-V5/WMFHomeNearbyCell.m

```diff
@@ -0,0 +1,175 @@
+
+#import "WMFHomeNearbyCell.h"
+#import "WMFCompassView.h"
+
+#import "Wikipedia-Swift.h"
+#import "PromiseKit.h"
+
+#import <Masonry/Masonry.h>
+
+static CGFloat const WMFTextPadding    = 8.0;
+static CGFloat const WMFDistanceHeight = 20.0;
+
+static CGFloat const WMFImageSize    = 104;
+static CGFloat const WMFImagePadding = 8.0;
+
+@interface WMFHomeNearbyCell ()
+
+@property (strong, nonatomic) IBOutlet UIImageView* imageView;
+@property (strong, nonatomic) IBOutlet WMFCompassView* compassView;
+@property (strong, nonatomic) IBOutlet UILabel* titleLabel;
+@property (strong, nonatomic) IBOutlet UIView* distanceLabelBackground;
+@property (strong, nonatomic) IBOutlet UILabel* distanceLabel;
+
+@end
+
+@implementation WMFHomeNearbyCell
+
+- (void)prepareForReuse {
+    [super prepareForReuse];
+    [[WMFImageController sharedInstance] cancelFetchForURL:self.imageURL];
+    self.imageView.image = [UIImage imageNamed:@"logo-placeholder-nearby.png"];
+    _imageURL            = nil;
+    _titleLabel.text     = nil;
+    _distanceLabel.text  = nil;
+    _titleText           = nil;
+    _descriptionText     = nil;
+}
+
+- (void)awakeFromNib {
+    [super awakeFromNib];
+    self.imageView.image                            = [UIImage imageNamed:@"logo-placeholder-nearby.png"];
+    self.imageView.layer.cornerRadius               = self.imageView.bounds.size.width / 2;
+    self.imageView.layer.borderWidth                = 1.0 / [UIScreen mainScreen].scale;
+    self.imageView.layer.borderColor                = [UIColor colorWithWhite:0.9 alpha:1.0].CGColor;
+    self.distanceLabelBackground.layer.cornerRadius = 2.0;
+}
+
+- (UICollectionViewLayoutAttributes*)preferredLayoutAttributesFittingAttributes:(UICollectionViewLayoutAttributes*)layoutAttributes {
+    self.titleLabel.preferredMaxLayoutWidth = layoutAttributes.size.width - WMFImageSize - WMFImagePadding - WMFImagePadding;
+    UICollectionViewLayoutAttributes* preferredAttributes = [layoutAttributes copy];
+    CGFloat height                                        = MAX(120, self.titleLabel.intrinsicContentSize.height + WMFTextPadding + WMFTextPadding + WMFDistanceHeight + WMFTextPadding);
+    preferredAttributes.size = CGSizeMake(layoutAttributes.size.width, height);
+    return preferredAttributes;
+}
+
+- (void)setImageURL:(NSURL*)imageURL {
+    _imageURL = imageURL;
+    @weakify(self);
+    [[WMFImageController sharedInstance] fetchImageWithURL:imageURL]
```

Comment:
cancel previous fetch first. also, might as well try to set from memory cache before fetching